llrt_core/src/modules/http/headers.rs

```diff
@@ -122,12 +160,24 @@ impl Headers {
     }
 
     pub fn from_map(map: BTreeMap<String, Coerced<String>>) -> Self {
-        let headers: BTreeMap<String, String> = map
+        let headers = map
             .into_iter()
-            .map(|(key, value)| (key.to_lowercase(), value.to_string()))
+            .map(|(k, v)| (k.to_lowercase(), v.to_string()))
             .collect();
         Self { headers }
     }
+
+    fn array_to_headers(array: Array<'_>) -> Result<Vec<(String, String)>> {
+        let mut vec = Vec::new();
+        for entry in array.into_iter().flatten() {
+            if let Some(array_entry) = entry.as_array() {
+                let key = array_entry.get::<String>(0)?;
+                let value = array_entry.get::<String>(1)?;
+                vec.push((key, value));
```

Comment:
Don't forget to lowercase key

Reply:
Thanks for pointing that out. I will correct it.